src/EditorFeatures/CSharpTest/InlineMethod/CSharpInlineMethodTests.cs

```diff
@@ -0,0 +1,2379 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InlineMethod;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.InlineMethod
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsInlineMethod)]
+    public class CSharpInlineMethodTests
+    {
+        private class TestVerifier : CSharpCodeRefactoringVerifier<CSharpInlineMethodRefactoringProvider>.Test
+        {
+            public static async Task TestInRegularAndScript1Async(
+                string initialMarkUp,
+                string expectedMarkUp,
+                bool keepInlinedMethod = true)
+            {
+                var test = new TestVerifier { CodeActionIndex = keepInlinedMethod ? 0 : 1 };
+                test.TestState.Sources.Add(initialMarkUp);
+                test.FixedState.Sources.Add(expectedMarkUp);
+                await test.RunAsync().ConfigureAwait(false);
+            }
+        }
+
+        [Fact]
+        public Task TestInlineInvocationExpressionForExpressionStatement()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+using System.Collections.Generic;
+public class TestClass
+{
+    private void Caller(int i)
+    {
+        var h = new HashSet<int>();
+        Ca[||]llee(i, h);
+    }
+
+    private bool Callee(int i, HashSet<int> set)
+    {
+        return set.Add(i);
+    }
+}",
+                @"
+using System.Collections.Generic;
+public class TestClass
+{
+    private void Caller(int i)
+    {
+        var h = new HashSet<int>();
+        h.Add(i);
+    }
+
+    private bool Callee(int i, HashSet<int> set)
+    {
+        return set.Add(i);
+    }
+}");
+
+        [Fact]
+        public Task TestRemoveTheInlinedMethod()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+        Ca[||]llee(i, j);
+    }
+
+    private void Callee(int i, int j)
+    {
+        System.Console.WriteLine(i + j);
+    }
+}",
+                @"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+        System.Console.WriteLine(i + j);
+    }
+}", keepInlinedMethod: false);
```

Comment:
so, IMO, it would have probably been better here to just have a single test method test both cases.  so you'd pass three strings:

1. initial code
2. code action 1
3. code action 2

but i get that would be a lot of work to update this harness.

Reply:
Done